tests/core/contracts/test_contract_events_buildFilter.py

```diff
@@ -0,0 +1,17 @@
+import json
+
+CONTRACT_ABI = json.loads('[{"constant":false,"inputs":[],"name":"return13","outputs":[{"name":"result","type":"int256"}],"type":"function"},{"constant":true,"inputs":[],"name":"counter","outputs":[{"name":"","type":"uint256"}],"type":"function"},{"constant":false,"inputs":[{"name":"amt","type":"uint256"}],"name":"increment","outputs":[{"name":"result","type":"uint256"}],"type":"function"},{"constant":false,"inputs":[{"name":"a","type":"int256"},{"name":"b","type":"int256"}],"name":"add","outputs":[{"name":"result","type":"int256"}],"type":"function"},{"constant":false,"inputs":[],"name":"increment","outputs":[{"name":"","type":"uint256"}],"type":"function"},{"constant":false,"inputs":[{"name":"a","type":"int256"}],"name":"multiply7","outputs":[{"name":"result","type":"int256"}],"type":"function"},{"anonymous":false,"inputs":[{"indexed":false,"name":"value","type":"uint256"}],"name":"Increased","type":"event"}]')  # noqa: E501
+
+
+def test_build_filter(web3):
+    contract = web3.eth.contract(abi=CONTRACT_ABI)
+    filter_builder = contract.events.Increased.build_filter()
+    filter_builder.args['value'].match_any(100, 200, 300)
+    _filter = filter_builder.deploy(web3)
+    assert _filter.filter_params == {
+        'topics': (
+            '0x3496c3ede4ec3ab3686712aa1c238593ea6a42df83f98a5ec7df9834cfa577c5',)}
+    assert _filter.data_filter_set == (('uint256', (100, 200, 300)),)
+    filter_builder.args['value'].match_single(400)
```

Comment:
Ive gone back and forth on this.  Because the `deploy` method generates a new filter from the builder's settings, it seemed not such a risk to allow the builder to be mutable.  But it is probably better to have the first release take the least mutable approach because taking away some mutability would be a breaking change whereas adding it would not.